python-3.14 has exports.h and it's include path is in front of this package's
exports.h

--- qt_gui_cpp/src/qt_gui_cpp/ros_pluginlib_plugin_provider.cpp~	2020-07-24 17:59:00.000000000 +0200
+++ qt_gui_cpp/src/qt_gui_cpp/ros_pluginlib_plugin_provider.cpp	2026-09-08 11:14:00.254270988 +0200
@@ -32,7 +32,7 @@
 #include "plugin_provider.h"
 #include "plugin.h"
 #include "ros_pluginlib_plugin_provider.h"
-#include "exports.h"
+#include "qt_gui_cpp/exports.h"
 
 // Explicit instantiation with export declaration for Windows.
 template class QT_GUI_CPP_DECL qt_gui_cpp::RosPluginlibPluginProvider<qt_gui_cpp::Plugin>;
